source/learn/getting-started/auth-with-rom.html.md

```diff
@@ -0,0 +1,269 @@
+---
+chapter: Getting Started
+title: Simple authentication with ROM and Rails
+---
+
+## Initial setup
+
+This guide describes how to add a simplest form of user authentication using ROM, Warden and Rails.
+
+> Please note that this guide is only for illustrative purposes and does not include validations 
+
+First of all, let's add following gems to the Gemfile:
+
+```ruby
+# Gemfile
+gem 'rom-rails'
+gem 'rom-sql'
+gem 'rom-repository'
+gem 'rom'
+gem 'warden'
+gem 'bcrypt'
+gem 'sqlite3'
+```
+
+Now we will need to add configuration for ROM to work. In this example we are going to use SQLite3 to store our data:
+
+```ruby
+# config/initializers/rom.rb
+ROM::Rails::Railtie.configure do |config|
+  config.gateways[:default] = [:sql, "sqlite://db/dev.db"]
+end
+```
+
+After this, we will need to add ROM rake tasks in our Procfile
+
+```ruby
+# Procfile
+require 'rom/sql/rake_task'
+```
+
+When this is done, let's generate a migration that adds users table to our database:
+
+```bash
+rake db:create_migration[create_users]
+# <= migration file created db/migrate/20161109173831_create_users.rb
+```
+
+Let's open up our migration file and add the following lines:
+
+```ruby
+# db/migrate/20161109173831_create_users.rb
+ROM::SQL.migration do
+  change do
+    create_table :users do
+      primary_key :id
+      column :email, String, null: false, unique: true
+      column :encrypted_password, String, null: false
+    end
+  end
+end
+```
+
+We want emails to be unique, and both password and email fields to not be NULL.
+
+Let's run our migration task:
+
+```bash
+rake db:migrate
+# <= db:migrate executed
+```
+
+Once that is done, we can start working on our repositories and rom components. Let's add schema file first:
```

Comment:
here as well, and in all other places, basically there must always be an empty line between a line of text and code block syntax